server-apollo/src/schema.js

```diff
@@ -49,18 +49,22 @@ const resolvers = {
       return result
     },
     updateUser: async (obj, args, context, info) => {
-      let { id, version, ...updateArgs } = args
-      const currentRecord = await context.db('users').select().where('id', args.id).then((rows) => rows[0])
-      if (!currentRecord) return null // or not found error??
+      return new Promise((resolve, reject) => {
+        context.db.transaction(async (trx) => {
+          let { id, version, ...updateArgs } = args
+          const currentRecord = await trx('users').select().where('id', args.id).then((rows) => rows[0])
+          if (!currentRecord) return null // or not found error??
 
-      const conflict = context.detectConflict(currentRecord, args) // detect conflict
+          const conflict = context.detectConflict(currentRecord, args) // detect conflict
 
-      if (conflict) {
-        updateArgs = context.handleConflict(context.conflictHandlers.RETURN_TO_CLIENT, conflict, currentRecord, args)
-      }
+          if (conflict) {
+            updateArgs = context.handleConflict(context.conflictHandlers.RETURN_TO_CLIENT, conflict, currentRecord, args)
+          }
 
-      const result = await context.db('users').update({ ...updateArgs, version: currentRecord.version + 1 }).where({ 'id': id }).returning('*').then((rows) => rows[0])
-      return result
+          const result = await trx('users').update({ ...updateArgs, version: currentRecord.version + 1 }).where({ 'id': id }).returning('*').then((rows) => rows[0])
```

Comment:
trx can be still context.db in this case. Is it possible to execute the same operations on `context.db` as on `trx` ?

Reply:
I don't think so. I'm guessing that whatever happens underneath is that the `trx` thing references back to the transaction. However, we would probably open the transaction and then provide the user resolver with the trx object :)

Reply:
So, put some thoughts into this. However users need to be able to perform select and update within transaction which is happening here. So ideally we should just expose trx as method and document that as knex transation.